exchanges/poloniex/poloniex.go

```diff
@@ -16,887 +16,1048 @@ import (
-	poloniexAvailableBalances    = "returnAvailableAccountBalances"
-	poloniexTradableBalances     = "returnTradableBalances"
-	poloniexTransferBalance      = "transferBalance"
-	poloniexMarginAccountSummary = "returnMarginAccountSummary"
-	poloniexMarginBuy            = "marginBuy"
-	poloniexMarginSell           = "marginSell"
-	poloniexMarginPosition       = "getMarginPosition"
-	poloniexMarginPositionClose  = "closeMarginPosition"
-	poloniexCreateLoanOffer      = "createLoanOffer"
-	poloniexCancelLoanOffer      = "cancelLoanOffer"
-	poloniexOpenLoanOffers       = "returnOpenLoanOffers"
-	poloniexActiveLoans          = "returnActiveLoans"
-	poloniexLendingHistory       = "returnLendingHistory"
-	poloniexAutoRenew            = "toggleAutoRenew"
-	poloniexCancelByIDs          = "/orders/cancelByIds"
-	poloniexTimestamp            = "/timestamp"
-	poloniexWalletActivity       = "/wallets/activity"
-	poloniexMaxOrderbookDepth    = 100
+	apiURL           = "https://api.poloniex.com"
+	tradeSpotPath    = "/trade/"
+	tradeFuturesPath = "/futures" + tradeSpotPath
+	marketsPath      = "/markets/"
 )
 
-// Exchange implements exchange.IBotExchange and contains additional specific api methods for interacting with Poloniex
+var (
+	errAddressRequired         = errors.New("address is required")
+	errInvalidWithdrawalChain  = errors.New("invalid withdrawal chain")
+	errInvalidTimeout          = errors.New("invalid timeout")
+	errChainsNotFound          = errors.New("chains not found")
+	errAccountIDRequired       = errors.New("missing account ID")
+	errAccountTypeRequired     = errors.New("account type required")
+	errMarginAdjustTypeMissing = errors.New("margin adjust type invalid")
+	errPositionModeInvalid     = errors.New("invalid position mode")
+	errTrailingOffsetInvalid   = errors.New("invalid trailing offset required for trailing stop orders")
+	errOffsetLimitInvalid      = errors.New("invalid offset required for trailing stop limit orders")
+)
+
+// Exchange is the overarching type across the poloniex package
 type Exchange struct {
 	exchange.Base
-	details CurrencyDetails
 }
 
-// GetTicker returns current ticker information
-func (e *Exchange) GetTicker(ctx context.Context) (map[string]Ticker, error) {
-	type response struct {
-		Data map[string]Ticker
+// GetSymbolInformation returns symbol and trade limit info
+func (e *Exchange) GetSymbolInformation(ctx context.Context, symbol currency.Pair) ([]*SymbolDetail, error) {
+	if symbol.IsEmpty() {
+		return nil, currency.ErrSymbolStringEmpty
 	}
+	var resp []*SymbolDetail
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, unauthEPL, "/markets/"+symbol.String(), &resp)
+}
 
-	resp := response{}
-	path := "/public?command=returnTicker"
+// GetAllSymbolInformation returns all symbol and trade limit info
+// symbol may be an empty currency.Pair to return all symbols
+func (e *Exchange) GetAllSymbolInformation(ctx context.Context) ([]*SymbolDetail, error) {
```

Comment:
🚧 Change request:
* Second comment is not true
* As above, removing `Information` and fixing `Details`
* Reworded first comment
```suggestion
// GetSymbols returns all symbols and thier trade limits
func (e *Exchange) GetSymbols(ctx context.Context) ([]*SymbolDetails, error) {
```